playbook.yml

```diff
@@ -0,0 +1,61 @@
+---
+- name: Playbook to install Taskmaster application
+  hosts: webservers
+  remote_user: ec2-user
+
+  vars_prompt:
+    - name: trello_api_key
+      prompt: What is the Trello API key?
+      private: true
+
+    - name: trello_token
+      prompt: What is the Trello token?
+      private: true 
+
+  tasks:
+    - name: Check git install
+      become: yes
+      ansible.builtin.yum: 
+       name: git
+       state: present
+    - name: Install latest Python
+      become: yes
+      ansible.builtin.yum: 
+       name: python3
+       state: latest
+    - name: Install Poetry
+      ansible.builtin.shell: 
+        cmd: curl -sSL https://install.python-poetry.org | python3
+        creates: /home/ec2-user/.local/bin/poetry
+    - name: Create the application home directory and change ownership
+      become: yes
+      ansible.builtin.command:
+        cmd: mkdir -p /opt/todoapp && chown ec2-user /opt/todoapp
+        creates: /opt/todoapp
+    - name: Clone repo of todo application from git
+      become: yes
+      ansible.builtin.git:
+        repo: 'https://github.com/ma7nas/DevOps-Course-Starter'
+        version: exercise-4
+        dest: /opt/todoapp
+    - name: Install Poetry dependencies
+      ansible.builtin.command:
+        chdir: /opt/todoapp
+        cmd: /home/ec2-user/.local/bin/poetry install
```

Comment:
Missed your question yesterday, but re having Ansible report `poetry install` more accurately, we could try using the `changed_when` option ([docs](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#defining-changed))

For example we might be able to do something similar to the example there:
```yml
- name: Combine multiple conditions to override 'changed' result
  ansible.builtin.command: /bin/fake_command
  register: result
  changed_when:
    - '"ERROR" in result.stderr'
```
I haven't tested, but for example:
```yml
  changed_when:
    - '!"No dependencies to install or update" in result.stdout'

```